x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ConstantNullBlock.java

```diff
@@ -68,6 +68,7 @@ public ElementType elementType() {
 
     @Override
     public Block filter(int... positions) {
+        close();
```

Comment:
This is so weird. Usually `filter` preserves a reference to the underlying block. But it doesn't for `ConstantNullBlock` and that's fine and good. But we presume that it does. This "simulates" that. I hate it.

Reply:
Yeah. this finds sucks. It's ok for now.